src/invoice_generator.py

```diff
@@ -1,54 +1,111 @@
 import datetime
+import importlib.resources
 import uuid
-import xml.etree.cElementTree as ET
-from typing import Dict
+from dataclasses import dataclass, asdict
 
+import jinja2
+import pdfkit
 
-def generate_invoice_xml(client_data: Dict, date: datetime.date | None = None, indent: str = "    ") -> str:
-    """
-    Generates an invoice in XML format. Creates a file named "invoice-{invoice_number}.xml" and returns its filename.
-    The invoice payment is due to the first day of the next month, relative to the invoice's date.
 
-    :param client_data: Data of the client for whom the invoice is to be generated. Must contain fields:
-        "name", "surname", "mail", "subscriptions".
-    :param date: Date of the invoice. If left as None, current date is used.
-    :param indent: Indentation of the XML file
-    :return: Filename of the generated invoice
-    """
+@dataclass
+class Invoice:
+    client_name: str
+    client_surname: str
+    client_mail: str
+    client_subscription: str
+    client_subscription_cost: float
+    client_packages: list[tuple[str, float]]
+    total_cost: float
+    invoice_date: datetime.date
+    payment_due: datetime.date
+    company_name: str = "Drying Paint Inc."
```

Comment:
there is already merged pull request containing file that included all necessary constants (manifest.py)